tests/unit/system/classes/MediaLibraryTest.php

```diff
@@ -62,4 +68,60 @@ public function testValidPathsOnValidatePath($path)
         $result = MediaLibrary::validatePath($path);
         $this->assertInternalType('string', $result);
     }
+
+    public function testListFolderContents()
+    {
+        $this->copyMedia();
+
+        $contents = MediaLibrary::instance()->listFolderContents();
+        $this->assertNotEmpty($contents, 'Media library item is not discovered');
+
+        $item = reset($contents);
+        $this->assertAttributeEquals('file', 'type', $item, 'Media library item does not have the right type');
+        $this->assertAttributeEquals('/text.txt', 'path', $item, 'Media library item does not have the right path');
+        $this->assertAttributeNotEmpty('lastModified', $item, 'Media library item last modified is empty');
+        $this->assertAttributeNotEmpty('size', $item, 'Media library item size is empty');
+
+        $this->removeMedia();
```

Comment:
@DieterHolvoet you will need to make this part of the `tearDown` process in the test class - otherwise, if the test fails, the test files that you had copied over will still be stored on disk.

Reply:
Yes, the storage path is changed by calling
```php
$this->app->useStoragePath(base_path('storage/temp'));
```

Reply:
This causes errors in the path validation tests. Also, the files will be copied and deleted 26 times because of those path validation tests, which seems a little wasteful. Is it an option to move this new test to a separate class?

Reply:
@DieterHolvoet Alright, well then the `setupStorage` method should be run only in this test case, and not as part of the `setUp` callback. It's not needed for those other tests.

I'd prefer your new test to be kept in this class as it is testing a part of the MediaLibrary class.

Reply:
@bennothommo Did the changes, let me know what you think.

Reply:
Thanks @DieterHolvoet - looks fine to me.